components/threatconnect.js

```diff
@@ -724,8 +842,250 @@ polarity.export = PolarityComponent.extend({
     },
     getField(indicatorId, field) {
       this.getField(indicatorId, field);
+    },
+    setIntegrationSelection(caseObj) {
+      let integrationData = this.getIntegrationData();
+      let annotations = this.getAnnotations();
+
+      if (Array.isArray(annotations) && annotations.length > 0) {
+        integrationData.unshift({
+          integrationName: 'Polarity Annotations',
+          data: annotations,
+          selected: false,
+          isAnnotations: true
+        });
+      }
+
+      if (!caseObj.__state) {
+        Ember.set(caseObj, '__state', {});
+      }
+
+      Ember.set(caseObj.__state, 'integrations', integrationData);
+      Ember.set(caseObj.__state, 'numSelectedWriteIntegrations', this.getNumSelectedIntegration(integrationData));
+    },
+    toggleAllIntegrations(caseObj) {
+      if (!caseObj.__state) {
+        Ember.set(caseObj, '__state', {
+          integrations: [],
+          numSelectedWriteIntegrations: 0,
+          showIntegrationData: false,
+          spinRefresh: false
+        });
+      }
+
+      const integrations = caseObj.__state.integrations || [];
+      const hasUnselected = integrations.some((int) => !int.selected);
+
+      integrations.forEach((integration) => {
+        Ember.set(integration, 'selected', hasUnselected);
+      });
+
+      this.setNumSelectedIntegrationsForCase(caseObj);
+    },
+    addNoteIntegrationSelected(caseObj) {
+      this.setNumSelectedIntegrations(caseObj);
+    },
+    toggleCaseIntegrationCheckbox(caseObj, event) {
+      const checked = event.target.checked;
+
+      Ember.set(caseObj.__state, 'showIntegrationData', checked);
+
+      if (checked) {
+        this.refreshIntegrationsForCase(caseObj);
+      }
+    },
+    createNote(caseObj) {
+      const state = caseObj.__state;
+
+      const includeIntegrationData = state.showIntegrationData;
+      let selectedIntegrations = [];
+      let annotations;
+
+      if (state.integrations) {
+        selectedIntegrations = state.integrations.filter(
+          (integration) => integration.selected && !integration.isAnnotations
+        );
+        annotations = state.integrations.find((integration) => integration.selected && integration.isAnnotations);
+      }
+
+      if (includeIntegrationData && selectedIntegrations.length === 0 && !annotations) {
+        Ember.set(state, 'missingIntegrations', true);
+        return;
+      }
+
+      const payload = {
+        action: 'ADD_INTEGRATION_DATA_AS_NOTE',
+        caseId: caseObj.id,
+        mode: 'append',
+        includeIntegrationData
+      };
+
+      if (includeIntegrationData) {
+        payload.integrationData = selectedIntegrations;
+        payload.annotations = annotations;
+      }
+
+      this.sendIntegrationMessage(payload)
+        .then((result) => {
```

Comment:
We need to check if this `result` object has an `error` property on it with a `detail` message.  The server is currently sending an error message if adding notes is not enabled.  However, the component side doesn't check for this and tells the user the note was created successfully even if the note was not created (due to the "Enable Adding Notes" option being disabled.